src/portals/hope/features/member/hooks/use-member-claim-payments.ts

```diff
@@ -0,0 +1,227 @@
+import { addYears } from 'date-fns'
+import gql from 'graphql-tag'
+import { useMemo } from 'react'
+import { useMemberClaimPaymentsQuery } from 'types/generated/graphql'
+import { InsuranceType } from '../../config/constants'
+import { getOneYearMasterInceptionBuckets } from '../../contracts/utils'
+
+gql`
+  query MemberClaimPayments($memberId: ID!) {
+    member(id: $memberId) {
+      memberId
+      claims {
+        id
+        subclaims {
+          contract {
+            insuranceType
+          }
+          paymentOrders {
+            amount
+            deductible
+            settledAt
+          }
+        }
+      }
+    }
+  }
+`
+
+export const useMemberClaimPayments = ({ memberId }: { memberId: string }) => {
+  const { data: { member } = {} } = useMemberClaimPaymentsQuery({
+    variables: { memberId },
+  })
+
+  const insuranceTypePayments = useMemo(() => {
+    if (!member) return undefined
+    return member.claims.reduce((acc, claim) => {
+      const claimPaymentPerInsuranceType = claim.subclaims.reduce(
+        (payments, subclaim) => {
+          const insuranceType = subclaim?.contract
+            ?.insuranceType as InsuranceType
+          if (!insuranceType) return payments
+          if (payments[insuranceType]) {
+            payments[insuranceType] = [
+              ...payments[insuranceType],
+              ...subclaim.paymentOrders
+                .filter(({ settledAt }) => Boolean(settledAt))
+                .map(
+                  ({
+                    amount: { amount, currency },
+                    deductible: { amount: deductible },
+                    settledAt: date,
+                  }) => ({
+                    amount,
+                    deductible,
+                    currency,
+                    date: new Date(date),
+                  }),
+                ),
+            ]
+          } else {
+            payments[insuranceType] = subclaim.paymentOrders
+              .filter(({ settledAt }) => Boolean(settledAt))
+              .map(
+                ({
+                  amount: { amount, currency },
+                  deductible: { amount: deductible },
+                  settledAt: date,
+                }) => ({
+                  amount,
+                  deductible,
+                  currency,
+                  date: new Date(date),
+                }),
+              )
+          }
+          return payments
+        },
```

Comment:
This should work or? 
```suggestion
            payments[insuranceType] = [
              ...(payments[insuranceType] ?? []),
              ...subclaim.paymentOrders
                .filter(({ settledAt }) => Boolean(settledAt))
                .map(
                  ({
                    amount: { amount, currency },
                    deductible: { amount: deductible },
                    settledAt: date,
                  }) => ({
                    amount,
                    deductible,
                    currency,
                    date: new Date(date),
                  }),
                ),
            ]
          return payments
        },
```

Reply:
Of course! 🙏